omod-2.2/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs2_2/ConditionResource2_2.java

```diff
@@ -0,0 +1,214 @@
+	 */
+	@Override
+	public DelegatingResourceDescription getRepresentationDescription(Representation representation) {
+		if (representation instanceof DefaultRepresentation) {
+			DelegatingResourceDescription description = new DelegatingResourceDescription();
+			description.addProperty("uuid");
+			description.addProperty("display");
+			description.addProperty("condition");
+			description.addProperty("clinicalStatus");
+			description.addProperty("verificationStatus");
+			description.addProperty("previousVersion");
+			description.addProperty("onsetDate");
+			description.addProperty("voided");
+			description.addSelfLink();
+			description.addLink("full", ".?v=" + RestConstants.REPRESENTATION_FULL);
+			return description;
+		} else if (representation instanceof FullRepresentation) {
+			DelegatingResourceDescription description = new DelegatingResourceDescription();
+			description.addProperty("uuid");
+			description.addProperty("display");
+			description.addProperty("condition");
+			description.addProperty("clinicalStatus");
+			description.addProperty("verificationStatus");
+			description.addProperty("previousVersion");
+			description.addProperty("onsetDate");
+			description.addProperty("additionalDetail");
+			description.addProperty("voided");
+			description.addProperty("auditInfo");
+			description.addSelfLink();
+			return description;
+		}
+		return null;
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceHandler#getGETModel(Representation)
+	 */
+	public Model getGETModel(Representation rep) {
+		ModelImpl model = ((ModelImpl) super.getGETModel(rep));
+		if (rep instanceof DefaultRepresentation || rep instanceof FullRepresentation) {
+			model
+			        .property("uuid", new StringProperty())
+			        .property("display", new StringProperty())
+			        .property("condition", new StringProperty())
+			        .property("clinicalStatus", new StringProperty())
+			        .property("verificationStatus", new StringProperty())
+			        .property("previousVersion", new StringProperty())
+			        .property("onsetDate", new StringProperty())
+			        .property("voided", new StringProperty());
+		}
+		return model;
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceHandler#getCREATEModel(Representation)
+	 */
+	@Override
+	public Model getCREATEModel(Representation rep) {
+		return new ModelImpl()
+		        .property("condition", new StringProperty())
+		        .property("patient", new StringProperty())
+		        .property("clinicalStatus", new StringProperty())
+		        .property("verificationStatus", new StringProperty())
+		        .property("previousVersion", new StringProperty())
+		        .property("onsetDate", new StringProperty())
+		        .required("condition").required("onsetDate");
```

Comment:
What does the above line mean?

Reply:
It specifies the required properties that must be set